jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildImageTask.java

```diff
@@ -71,11 +95,93 @@ public boolean getReproducible() {
     return format;
   }
 
+  @Input
+  public boolean getUseOnlyProjectCache() {
+    return useOnlyProjectCache;
+  }
+
   @TaskAction
-  public void buildImage() {
-    // TODO: Implement.
+  public void buildImage() throws InvalidImageReferenceException, IOException {
+    // Asserts required @Input parameters are not null.
+    Preconditions.checkNotNull(from);
+    Preconditions.checkNotNull(from.getImage());
+    Preconditions.checkNotNull(to);
+    Preconditions.checkNotNull(to.getImage());
+    Preconditions.checkNotNull(jvmFlags);
+    Preconditions.checkNotNull(format);
+
+    ImageReference baseImageReference = ImageReference.parse(from.getImage());
+    ImageReference targetImageReference = ImageReference.parse(to.getImage());
+
+    ProjectProperties projectProperties = new ProjectProperties(getProject(), getLogger());
+
+    String mainClass = this.mainClass;
+    if (mainClass == null) {
```

Comment:
were we going to do that ${main-class} check thing here too?

Reply:
The one regarding the unresolved ${start-class} in spring? That has not been implemented in jib-maven-plugin either yet.

Reply:
ah okay, lets track that.